codalab/codalab-worksheets
# Instrument BundleCLI with autocomplete

Fixes #167 
Involves a major refactor of the argparse configuration. Apologies in advance, since this was necessary to build an instance of `ArgumentParser` well-formed enough to do autocompletion inference.

To test it out:
1.  Checkout the `autocomplete` branch
2.  Rerun `setup.sh`
3.  Make sure you are sourcing `rc` in your `~/bashrc`
4.  If you use `zsh`, read the comments in `rc`.

Currently only worksheet specs are being autocompleted (in addition to subcommands and flags), but please take a look at the new code structure etc. and test it out when you have time while I'm still implementing the other completers, since this is a rather hefty diff. @percyliang @pbhatnagar3
